pkg/sync/syncer.go

```diff
@@ -1967,8 +2059,6 @@ func (s *syncer) processGrantsWithExternalPrincipals(ctx context.Context, princi
 		}
 	}
 
-	s.state.FinishAction(ctx)
```

Comment:
Where do we finish this action now? Or were we not supposed to finish it?

Reply:
the callers of this were already calling `s.state.FinishAction(ctx)`, so this was a bug that would finish the expansion action before we did it